pallets/funding/src/settlement.rs

```diff
@@ -0,0 +1,306 @@
+use super::*;
+
+use frame_support::{
+	dispatch::{ DispatchResult},
+	ensure,
+	pallet_prelude::*,
+	traits::{
+		fungible::{InspectHold, MutateHold as FungibleMutateHold},
+		fungibles::{
+            Inspect, Mutate as FungiblesMutate,
+		},
+		tokens::{Fortitude, Precision, Preservation, Restriction},
+		Get,
+	},
+};
+use sp_runtime::{traits::Zero, Perquintill};
+use polimec_common::{
+	migration_types::{MigrationInfo, MigrationOrigin, Migrations, ParticipationType},
+	ReleaseSchedule,
+};
+
+impl<T: Config> Pallet<T> {
+    pub fn do_settlement_success_bidder(bid: BidInfoOf<T>, project_id: ProjectId) -> DispatchResult {
+        
+        let project_details = ProjectsDetails::<T>::get(project_id).ok_or(Error::<T>::ProjectDetailsNotFound)?;
+		// Ensure that:
+        // 1. The project is in the FundingSuccessful state
+        // 2. The bid is in the Accepted or PartiallyAccepted state
+        // 3. The contribution token exists
+        ensure!(project_details.status == ProjectStatus::FundingSuccessful, Error::<T>::NotAllowed);
+        ensure!(matches!(bid.status, BidStatus::Accepted | BidStatus::PartiallyAccepted(..)), Error::<T>::NotAllowed);
+        ensure!(T::ContributionTokenCurrency::asset_exists(project_id), Error::<T>::CannotClaimYet);
```

Comment:
Just to be aligned, this check is really for making sure settlements only happen after `T::SuccessToSettlementTime`. We check the CT specifically since it is created when the settlement starts.